homeassistant/components/music_assistant/media_player.py

```diff
@@ -0,0 +1,568 @@
+    mass = entry.runtime_data.mass
+    added_ids = set()
+
+    async def handle_player_added(event: MassEvent) -> None:
+        """Handle Mass Player Added event."""
+        if TYPE_CHECKING:
+            assert event.object_id is not None
+        if event.object_id in added_ids:
+            return
+        added_ids.add(event.object_id)
+        async_add_entities([MusicAssistantPlayer(mass, event.object_id)])
+
+    # register listener for new players
+    entry.async_on_unload(mass.subscribe(handle_player_added, EventType.PLAYER_ADDED))
+    mass_players = []
+    # add all current players
+    for player in mass.players:
+        added_ids.add(player.player_id)
+        mass_players.append(MusicAssistantPlayer(mass, player.player_id))
+
+    async_add_entities(mass_players)
+
+
+class MusicAssistantPlayer(MusicAssistantEntity, MediaPlayerEntity):
+    """Representation of MediaPlayerEntity from Music Assistant Player."""
+
+    _attr_name = None
+    _attr_media_image_remotely_accessible = True
+    _attr_media_content_type = HAMediaType.MUSIC
+
+    def __init__(self, mass: MusicAssistantClient, player_id: str) -> None:
+        """Initialize MediaPlayer entity."""
+        super().__init__(mass, player_id)
+        self._attr_icon = self.player.icon.replace("mdi-", "mdi:")
+        self._attr_supported_features = SUPPORTED_FEATURES
+        if PlayerFeature.SYNC in self.player.supported_features:
+            self._attr_supported_features |= MediaPlayerEntityFeature.GROUPING
+        self._attr_device_class = MediaPlayerDeviceClass.SPEAKER
+        self._prev_time: float = 0
+
+    async def async_added_to_hass(self) -> None:
+        """Register callbacks."""
+        await super().async_added_to_hass()
+
+        # we subscribe to player queue time update but we only
+        # accept a state change on big time jumps (e.g. seeking)
+        async def queue_time_updated(event: MassEvent) -> None:
+            if event.object_id != self.player.active_source:
+                return
+            if abs((self._prev_time or 0) - event.data) > 5:
+                await self.async_on_update()
+                self.async_write_ha_state()
+            self._prev_time = event.data
+
+        self.async_on_remove(
+            self.mass.subscribe(
+                queue_time_updated,
+                EventType.QUEUE_TIME_UPDATED,
+            )
+        )
+
+    @property
+    def active_queue(self) -> PlayerQueue | None:
+        """Return the active queue for this player (if any)."""
+        if not self.player.active_source:
+            return None
+        return self.mass.player_queues.get(self.player.active_source)
+
+    @property
+    def extra_state_attributes(self) -> Mapping[str, Any]:
+        """Return additional state attributes."""
+        return {
+            ATTR_MASS_PLAYER_TYPE: self.player.type.value,
+            ATTR_ACTIVE_QUEUE: self.active_queue.queue_id
+            if self.active_queue
+            else None,
```

Comment:
Please wrap multiline lambdas in parenthesis

Reply:
done